test/box-luatest/downgrade_test.lua

```diff
@@ -783,3 +770,51 @@ g.test_downgrade_user_auth_history_and_last_modified = function(cg)
         check_NO_new_auth_fields()
     end)
 end
+
+-- Check that SQL CHECK constraints could be downgraded.
+g.test_downgrade_sql_tuple_check_contraints = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        box.execute([[CREATE TABLE t(i INT PRIMARY KEY, a INT);]])
+        box.execute([[ALTER TABLE t ADD CONSTRAINT c CHECK (a > 10);]])
+
+        local function constraint_before()
+            local space = box.space._space.index.name:get{'T'}
+            t.assert(space.flags.constraint ~= nil)
+            local func_id = space.flags.constraint.C
+            t.assert(func_id ~= nil)
+            local func = box.space._func:get{func_id}
+            t.assert(func ~= nil)
+            t.assert_equals(func.body, 'a > 10')
+        end
+
+        local function constraint_after(func_id)
+            local space = box.space._space.index.name:get{'T'}
```

Comment:
Nit. We can just use closure instead of argument.

Reply:
We can, but in this case we must either move the definition of `constraint_after()` after the first execution of `constraint_before()`, or initialize `func_id` before properly checking for the existence of the constraint. Both options do not look great for me.